### PR TITLE
[IDSEQ-1761] Description field is not saved when you collapse and re-open the side-bar.

### PR DESCRIPTION
**Pull request by ovalenzuela19**
# Description
Description field is not saved when you collapse and re-open the side-bar.

*The intended change, motivation and consequences.*
# Notes
The state in the parent component DiscoveryView was not getting the updated changes that was caused by saving the ProjectDescription component. ProjectDescription was only saving the changes to the server.

*Optional observations, comments or explanations.*
ProjectDescription component has a function called _handleDescriptionSave_ but may be misleading since it only saves the description to the server and the current component state... Since it does not update the state of _project_ in the DiscoveryView component, upon re-opening the side bar the description is not updated and thus shows the old description, not the updated one.

# Tests

* *Server-side code should have automated tests*
* *Client-side code and scripts should have at least a manual test plan*
* *See also [IDseq PR Guidelines](https://github.com/chanzuckerberg/idseq-web/blob/master/PR_GUIDELINES.md)*

### CODE REVIEW

File: app/assets/src/components/views/discovery/DiscoverySidebar/DiscoverySidebar.jsx [hunk 1]
```diff
@@ -105,7 +105,7 @@ export default class DiscoverySidebar extends React.Component {
       <div className={cs.histogramContainer}>
         <div className={cs.dateHistogram}>
           {dates.map(entry => {
-            const percent = Math.round(100 * entry.count / total, 0);
+            const percent = Math.round((100 * entry.count) / total, 0);
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/discovery/DiscoverySidebar/DiscoverySidebar.jsx:108]: Why the extra parentheses?

File: app/assets/src/components/views/discovery/DiscoverySidebar/ProjectDescription.jsx [hunk 2]
```diff
@@ -53,6 +53,7 @@ class ProjectDescription extends React.Component {
         },
         () => {
           this._save(this.props.project.id, this.state.description);
+          this.props.onProjectDescriptionSave(this.state.description);
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/discovery/DiscoverySidebar/ProjectDescription.jsx:56]: nit: Although this a simple function and the pattern was not followed before here, we try to use a pattern of loading `props` and `state` at the top of the function like:
```javascript
const { changed } = this.state;
const { onProjectDescriptionSave } = this.props;
```
Do you mind applying this here and the other functions you touched?

**Comment by ovalenzuela19** [comment]
I saw that there were unintentional changes to the files I edited and just changed the formatting of some code... am I missing a linter? Not sure why those lines were changed

File: app/assets/src/components/views/discovery/DiscoverySidebar/ProjectDescription.jsx (hunk 2 above)
Comment (ovalenzuela19) [reply app/assets/src/components/views/discovery/DiscoverySidebar/ProjectDescription.jsx:56]: Got it! I will change rn

File: app/assets/src/components/views/discovery/DiscoverySidebar/DiscoverySidebar.jsx (hunk 1 above)
Comment (ovalenzuela19) [reply app/assets/src/components/views/discovery/DiscoverySidebar/DiscoverySidebar.jsx:108]: I saw that there were unintentional changes to the files I edited and just changed the formatting of some code... am I missing a linter? Not sure why those lines were changed

Comment (tfrcarvalho) [reply app/assets/src/components/views/discovery/DiscoverySidebar/DiscoverySidebar.jsx:108]: Could be... or for some reason you are running a different linter version (globally installed?)  that has different default rules.

File: app/assets/src/components/views/discovery/DiscoverySidebar/ProjectDescription.jsx (hunk 2 above)
Comment (ovalenzuela19) [reply app/assets/src/components/views/discovery/DiscoverySidebar/ProjectDescription.jsx]: fixed!

**Comment by tfrcarvalho** [comment]
Congrats on your first PR merge! 👍

**Comment by MarkAZhang** [comment]
Congrats on your first PR, @ovalenzuela19!

**Comment by cdebourcy** [comment]
Hoorray!!

**Comment by lvreynoso** [comment]
Congrats!

**Pull request merged**